core/trino-spi/src/main/java/io/trino/spi/Preconditions.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi;
+
+import static java.lang.String.format;
+
+/**
+ * This class provides a very limited subset of methods found in Guava's Preconditions class.
+ * The SPI cannot be dependent on Guava since it is too heavy.
+ */
+public final class Preconditions
```

Comment:
I think we shouldn't expose these methods to outside of `trino-spi`.

Reply:
That's a good point.
Is there a way to restrict the usage to only a set of packages? Probably not (since we're not using java modules)

Reply:
Can we make it package private if it's going to be used only in trino-spi ?

Reply:
Indeed. Those methods were all private because we don’t want to expose this functionality to users of the SPI.

Reply:
It appears that majority of usages is in inner packages, so package-private will not work. I guess we can either:
- Make some verifier rule that prevents using this class from outside of SPI
- Close this PR

WDYT?

Reply:
True, I'll just close this then